### PR TITLE
feat(worker): launch without launching process

### PR DESCRIPTION
**Pull request by roggervalf**
re #436

### CODE REVIEW

File: src/classes/worker.ts [hunk 1]
```diff
@@ -69,6 +69,7 @@ export class Worker<
       concurrency: 1,
       lockDuration: 30000,
       runRetryDelay: 15000,
+      process: true,
```
Comment (manast) [inline src/classes/worker.ts:72]: I think "autorun" is a good name here actually.

File: src/classes/worker.ts [hunk 2]
```diff
@@ -102,13 +103,23 @@ export class Worker<
...
+        this.run().catch(error => {
+          this.running = false;
+          console.error(error);
+        });
+    }
+
+    this.on('error', err => console.error(err));
+  }
+
+  process(): void {
```
Comment (manast) [inline src/classes/worker.ts:116]: I think we can just make "run" public, so the error handling will be taken care by the caller. And inside the run() function we could just throw an exception if processFn is not defined (as it does not make sense to start the run loop if no process function is defined.

**Review by manast — CHANGES_REQUESTED** [review]
Good that you fixed this issue! Please check the comments.

File: src/classes/worker.ts (hunk 2 above)
Comment (roggervalf) [reply src/classes/worker.ts]: sure, let me refactor it a little bit

**Review by manast — COMMENTED** [review]
Looks good. I think you should squeeze this PR and use "feat" since you are adding a new feature.

**Pull request merged**